docs/testing/03-vitest.md

```diff
@@ -0,0 +1,183 @@
+---
+title: Vitest
+position: 5
+---
+
+# Overview
+
+[Vitest](https://vitest.dev/) is a popular and modern test framework for unit testing. You can use Vitest to test Stencil components in the browser using its [browser mode feature](https://vitest.dev/guide/browser.html).
+
+:::note
+Vitest browser mode is an experimental feature and in early development. As such, it may not yet be fully optimized, and there may be some bugs or issues that have not yet been ironed out.
+:::
+
+## Set Up
+
+To get started with Vitest, all you need to install it via:
+
+```bash npm2yarn
+npm install vitest @vitest/browser unplugin-stencil webdriverio
+```
+
+This command installs:
+
+- `vitest`: the core test framework
+- `@vitest/browser`: enables testing in browser environments
+- `unplugin-stencil`: integrates Stencil's compiler with Vitest for seamless testing
+- `webdriverio`: facilitates browser management for tests
+
+Next, we create a Vitest configuration as following:
+
+```ts vitest.config.ts
+import stencil from 'unplugin-stencil/vite'
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+    test: {
+        browser: {
+            enabled: true,
+            headless: true,
+            name: 'chrome'
+        },
+    },
+    plugins: [stencil()]
+})
+```
+
+This configuration enables tests to run in a headless Chrome browser.
+
+## Writing Tests
+
+Once you've setup Vitest you can start write your first test. In order to render a Stencil component into the browser, all you need to do is importing the component and initiating an instance of the component on the page:
+
+```ts
+// src/components/my-component/my-component.test.ts
```

Comment:
we could use the little `title` option like so if we want here:

````suggestion
```ts title="src/components/my-component/my-component.test.ts"
````

that goes for the other filenames in here as well